doc/architecture/layers.sql

```diff
@@ -0,0 +1,112 @@
+-- User table
+CREATE TABLE users (
+    ...
+);
+
+-- Project table
+CREATE TABLE map (
+    ...
+);
+
+-- Custom geometry type
+CREATE DOMAIN geom_type AS GEOMETRY
+CHECK (
+    GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+    ST_SRID(VALUE) = 4326
+);
+
+/*
+    Still up for discussion whether we want to also use cartesian coordinates
+
+    CREATE DOMAIN geom_cartesian_type AS GEOMETRY
+    CHECK (
+        GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+        ST_SRID(VALUE) = 3857
+    );
+ */
+
+-- Base Layer table
+CREATE TABLE base_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    base_type TEXT NOT NULL
+);
+
+-- Fertilize Layer table
+CREATE TABLE fertilize_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    fertilizer_type TEXT,
+    application_rate NUMERIC
+);
+
+-- Dimensioning Layer table
+CREATE TABLE dimensioning_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    dimension_type TEXT,
+    length NUMERIC,
+    width NUMERIC
+);
+
+-- Hydrology Layer table
+CREATE TABLE hydrology_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    water_feature_type TEXT,
```

Comment:
I presume this water_feature_type (as well as most of the [...]_type attributes) references a placeable element on the map? Is it planned, that they will also be foreign keys referencing the (not yet existing) table for holding all placable elements in the future?

Reply:
> I presume this water_feature_type (as well as most of the [...]_type attributes) references a placeable element on the map? 

You are correct! It is meant to store textual information about the specific type of placeable element on the map, such as "river", "lake", or "pond".

> Is it planned, that they will also be foreign keys referencing the (not yet existing) table for holding all placable elements in the future?

I was thinking about storing the elements related to a certain layer directly inside the according layer table, so it hasn't been planned yet.

Reply:
>  You are correct! It is meant to store textual information about the specific type of placeable element on the map, such as "river", "lake", or "pond".

Then it should be an enum of the available types.